tests/test_product.py

```diff
@@ -0,0 +1,64 @@
+import pytest
+from buy_together_app import models
+# from buy_together_app import views
+
+QR = "Q5o76MbdiNbXprNEnHfpcGWFp1CMF8XY"
+NAME = "Banana"
+DESCRIPTION = "Good!"
+
+
+@pytest.fixture
+def product0():
+    return models.Product(qr_code=QR, product_name=NAME, description=DESCRIPTION)
+
+
+class TestProductModel:
+    def test_new_product(self, product0):
+        assert product0.qr_code == QR
+        assert product0.product_name == NAME
+        assert product0.description == DESCRIPTION
+
+    @pytest.mark.django_db()
+    def test_save_product(self, product0):
+        product0.save()
+        # print(models.Product.objects.all()[0].product_name)
+        # assert False
+        assert product0 in models.Product.objects.all()
+
+    @pytest.mark.django_db()
+    def test_delete_product(self, product0):
+        product0.save()
+        # print(models.Product.objects.all())
+        # assert False
+        product0.delete()
+        assert product0 not in models.Product.objects.all()
+
+    # @pytest.mark.skip
+    @pytest.mark.django_db()
+    def test_filter_by_qr(self, product0):
+        product0.save()
+        # print(list(models.Product.filter_qr(product0.qr_code)))
+        # assert False
+        assert [product0] == list(models.Product.filter_qr(product0.qr_code))
+
+    # @pytest.mark.skip
+    @pytest.mark.django_db()
+    def test_filter_by_product_name(self, product0):
+        product0.save()
+        # print(list(models.Product.filter_name(product0.product_name))[0].product_name)
+        # assert False
+        assert product0 in list(models.Product.filter_name(product0.product_name))
+
+    # @pytest.mark.skip
+    @pytest.mark.django_db()
+    def test_filter_by_description(self, product0):
+        product0.save()
+        # print(list(models.Product.filter_description(product0.description)))
+        # assert False
+        assert product0 in models.Product.filter_description(product0.description)
+
+    # @pytest.mark.skip
+    @pytest.mark.django_db()
+    def test_get_all_products(self):
```

Comment:
Do you actually have any products there?

Reply:
iny my 0002 test file i have added 2 products and if there is none for example we dont get into the assert

Reply:
and how would you know you didn't get to the assert line? You need somehow make sure that this test is checking something.

Reply:
Fixed